src/cargo_communication/request_actor.rs

```diff
@@ -348,3 +348,837 @@ pub trait CargoHandleTrait<T> {
+                                .text("test_text".to_string())
+                                .highlight_start(0_usize)
+                                .highlight_end(0_usize)
+                                .build()
+                                .unwrap()])
+                            .label(Some("test_label".to_string()))
+                            .suggested_replacement(None)
+                            .suggestion_applicability(None)
+                            .expansion(None)
+                            .build()
+                            .unwrap(),
+                    ])
+                    .children(vec![DiagnosticBuilder::default()
+                        .message("test_child_message".to_string())
+                        .code(None)
+                        .level(DiagnosticLevel::Help)
+                        .spans(vec![DiagnosticSpanBuilder::default()
+                            .file_name("test_filename1".to_string())
+                            .byte_start(0_u32)
+                            .byte_end(0_u32)
+                            .line_start(5_usize)
+                            .line_end(6_usize)
+                            .column_start(7_usize)
+                            .column_end(8_usize)
+                            .is_primary(true)
+                            .text(vec![DiagnosticSpanLineBuilder::default()
+                                .text("test_text".to_string())
+                                .highlight_start(0_usize)
+                                .highlight_end(0_usize)
+                                .build()
+                                .unwrap()])
+                            .label(Some("test_label".to_string()))
+                            .suggested_replacement(None)
+                            .suggestion_applicability(None)
+                            .expansion(None)
+                            .build()
+                            .unwrap()])
+                        .children(vec![])
+                        .rendered(None)
+                        .build()
+                        .unwrap()])
+                    .rendered(Some("test_rendered".to_string()))
+                    .build()
+                    .unwrap(),
+            )
+            .build()
+            .unwrap();
+
+        send_from_cargo
+            .send(CargoStdout(CompilerMessage(compiler_mess)))
+            .unwrap();
+
+        let proper_publish_diagnostic_first = Message::Notification(Notification::new(
+            PublishDiagnostics::METHOD.to_string(),
+            PublishDiagnosticsParams {
+                text_document: TextDocumentIdentifier {
+                    uri: "file:///test_root_path/test_filename1".into(),
+                },
+                build_target: "".into(),
+                //TODO change to "test_target_name" later
+                origin_id: Some("test_origin_id".into()),
+                diagnostics: vec![
+                    LSPDiagnostic {
```

Comment:
"In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. "

https://rust-lang.github.io/api-guidelines/naming.html

Reply:
Poprawione